### PR TITLE
Feat(mysql): add LOCK property, allow properties after ALTER TABLE

### LINKED ISSUE #3020: Online DDL in Mysql
Could support for Mysql's online DDL please be added to Sqlglot? That is, DDL operations should support `ALGORITHM` and `LOCK` clauses.

See, e.g. https://dev.mysql.com/doc/refman/8.0/en/innodb-online-ddl.html

Here's an example of what I tried to do, the last step is what I'd like to see improved.

```
>>> import sys; sys.version_info
sys.version_info(major=3, minor=11, micro=7, releaselevel='final', serial=0)

>>> import sqlglot; sqlglot._version.version_tuple
(21, 1, 2)

>>> sqlglot.parse_one("ALTER TABLE t1 ADD COLUMN x INT;", dialect="mysql")
AlterTable(
  this=Table(
    this=Identifier(this=t1, quoted=False)),
  actions=[
    ColumnDef(
      this=Identifier(this=x, quoted=False),
      kind=DataType(this=Type.INT, nested=False))])

>>> sqlglot.parse_one("ALTER TABLE t1 ADD COLUMN x INT, ALGORITHM=INPLACE, LOCK=EXCLUSIVE;", dialect="mysql")
'ALTER TABLE t1 ADD COLUMN x INT, ALGORITHM=INPLACE, LOCK=EXCLUSIVE' contains unsupported syntax. Falling back to parsing as a 'Command'.
Command(this=ALTER, expression=TABLE t1 ADD COLUMN x INT, ALGORITHM=INPLACE, LOCK=EXCLUSIVE)
```

I'll take a look at the code myself but I suspect it will take me some time to understand how Sqlglot's modules are interrelated and the authors can probably do it faster/better.

Thanks!

### PR DESCRIPTION
**Pull request by georgesittas**
Fixes #3020

### CODE REVIEW

File: sqlglot/dialects/mysql.py [hunk 1]
```diff
@@ -408,6 +408,41 @@ class Parser(parser.Parser):
             "SPATIAL": lambda self: self._parse_index_constraint(kind="SPATIAL"),
         }
 
+        TABLE_OPTION_PARSERS = {
```
Comment (georgesittas) [inline sqlglot/dialects/mysql.py:411]: I'm wondering if we should make new property classes for these, wdyt @tobymao? I just realized some of them are already supported (e.g. `ALGORITHM` and `CHECKSUM`), so perhaps a `_parse_csv(self._parse_property)` could do the trick here if we created expression types for the unsupported ones.

Comment (tobymao) [reply sqlglot/dialects/mysql.py:411]: not sure what you mean

**Review by tobymao — APPROVED** [review]

File: sqlglot/dialects/mysql.py (hunk 1 above)
Comment (georgesittas) [reply sqlglot/dialects/mysql.py:411]: We can already parse some of these table options, e.g.

https://github.com/tobymao/sqlglot/blob/9595240a1c0f0e5ace9f67f31564e5d5edb9a9d2/tests/dialects/test_mysql.py#L73

The corresponding AST for this is:

```python
>>> import sqlglot
>>> sqlglot.parse_one("CREATE TABLE z (a INT) ENGINE=InnoDB AUTO_INCREMENT=1 DEFAULT CHARACTER SET=utf8 COLLATE=utf8_bin COMMENT='x'", "mysql")
Create(
  this=Schema(
    this=Table(
      this=Identifier(this=z, quoted=False)),
    expressions=[
      ColumnDef(
        this=Identifier(this=a, quoted=False),
        kind=DataType(this=Type.INT, nested=False))]),
  kind=TABLE,
  properties=Properties(
    expressions=[
      EngineProperty(
        this=Identifier(this=InnoDB, quoted=False)),
      AutoIncrementProperty(
        this=Literal(this=1, is_string=False)),
      CharacterSetProperty(
        this=Var(this=utf8),
        default=True),
      CollateProperty(
        this=Identifier(this=utf8_bin, quoted=False)),
      SchemaCommentProperty(
        this=Literal(this=x, is_string=True))]))
```

I'll see if I can refactor my PR to get rid of the table option parsers and reuse `_parse_properties` or something.

Comment (georgesittas) [reply sqlglot/dialects/mysql.py]: https://github.com/tobymao/sqlglot/pull/3027/commits/dba33c04c7d0d03ef3442998cc7159a1df2540c3

**Pull request merged**